xocto/ranges.py

```diff
@@ -838,3 +838,14 @@ def period_is_fully_covered(period: Range[T], ranges: Sequence[Range[T]]) -> boo
     range_set = RangeSet[T]([period])
     range_set -= RangeSet[T](ranges)
     return not bool(range_set)
+
+
+def any_overlapping(ranges: Sequence[Range[T]]) -> bool:
```

Comment:
🐼 Could you add a docstring for this function?

Reply:
Added.